networkx/algorithms/community/demon.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+#    Copyright (C) 2015
+#    All rights reserved.
+#    BSD license.
+
+"""
+Flat Merge version of DEMON (Democratic Estimate of the Modular Organization of
+a Network) algorithm as described in:
+
+Michele Coscia, Giulio Rossetti, Fosca Giannotti, Dino Pedreschi:
+DEMON: a local-first discovery method for overlapping communities.
+KDD 2012:615-623
+
+DEMON uses a local-first approach to community discovery were each node
+democratically votes for the communities that it sees in its surroundings. It
+provides a deterministic, fully incremental method for community discovery that
+has a limited time complexity.
+
+Based on the implementation by Giulio Rossetti <giulio.rossetti@isti.cnr.it>:
```

Comment:
I contacted both Michele Coscia and Giulio Rossetti before starting with this. They were kind enough to license their implementation of DEMON with a MIT license so I could add it to networkx ([here](http://www.michelecoscia.com/wp-content/uploads/2015/05/demon_py.zip)). Should I put this somehow in the documentation or the comments?